### DISCUSSION TITLE
npm pack undefined@undefined

### DISCUSSION THREAD

**Discussion by aminnairi** [npm/feedback#288] [CLI] [answered]
If you try to run `npm pack --dry-run` on a package that has no `name` property in its `package.json` file, the output shows `undefined@undefined` for the package name and `undefined-undefined.tgz` for the filename.

I think there should be a guard for when the `name` property, and even `version` and `description` properties are not set whenever running `npm pack`.

What are your thoughts on this?

**Comment by MylesBorins** [comment, marked as answer]
@aminnairi this would be a great topic to open as an issue on https://github.com/npm/cli
